src/core/types.ts

```diff
@@ -75,6 +75,8 @@ export interface Notice {
   details: Record<string, Record<string, unknown>>,
   __breadcrumbs: BreadcrumbRecord[],
   afterNotify?: AfterNotifyHandler,
+  cause?: Error|Record<string, unknown>,
+  causes: Array<{class: string, message: string, backtrace: BacktraceFrame[]}>,
```

Comment:
What is the need for causes if you always have 1 cause? Or am I missing something here?

Reply:
It's the API spec. But I think you're right—it should probably be on the payload, not the notice object.

Reply:
On second thought, perhaps I'll keep it as is🤔. It's a bit messy because we use the `Notice` type to represent both what we get (possibly an Error object) and what we produce from that. The `cause` property is from the Error object. The `causes` is the resolved value for the API, which we need to generate earlier, because of all the functions and transformations the notice object passes through.

Reply:
But you've reminded me that it might not be a bad idea to expand causes (ie follow the `.cause`'s cause, and so on. I've updated it to do this now. so there may be more than one cause.